drivers/sensors/lsm6dso32_uorb.c

```diff
@@ -0,0 +1,1979 @@
+
+struct lsm6dso32_sens_s
+{
+  FAR struct sensor_lowerhalf_s lower; /* Lower-half sensor driver */
+  FAR struct lsm6dso32_dev_s *dev;     /* Reference to parent device */
+  bool enabled;                        /* If this sensor is enabled */
+  enum lsm6dso32_odr_e odr;            /* Measurement interval of this
+                                        * sensor */
+  int fsr;                             /* Full scale range of this sensor.
+                                        * Can be from either gyro or accel
+                                        * FSR enum. */
+  sem_t run;                           /* Polling cycle lock */
+  enum lsm6dso32_int_e intpin;         /* The interrupt pin for this device */
+  bool interrupts;                     /* Whether or not interrupts are
+                                        * enabled. */
+  struct work_s work;                  /* Interrupt work queue
+                                        * structure */
+};
+
+/* Represents the LSM6DSO23 IMU device */
+
+struct lsm6dso32_dev_s
+{
+  struct lsm6dso32_sens_s gyro;  /* Gyroscope */
+  struct lsm6dso32_sens_s accel; /* Accelerometer lower half */
+  FAR struct i2c_master_s *i2c;  /* I2C interface. */
+  uint8_t addr;                  /* I2C address. */
+  float gy_off[3];               /* Offsets for gyroscope measurements */
+  mutex_t devlock;
+};
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static int lsm6dso32_control(FAR struct sensor_lowerhalf_s *lower,
+                             FAR struct file *filep, int cmd,
+                             unsigned long arg);
+static int lsm6dso32_activate(FAR struct sensor_lowerhalf_s *lower,
+                              FAR struct file *filep, bool enable);
+static int lsm6dso32_set_interval(FAR struct sensor_lowerhalf_s *lower,
+                                  FAR struct file *filep,
+                                  FAR uint32_t *period_us);
+static int lsm6dso32_selftest(FAR struct sensor_lowerhalf_s *lower,
+                              FAR struct file *filep, unsigned long arg);
+static int lsm6dso32_set_calibvalue(FAR struct sensor_lowerhalf_s *lower,
+                                    FAR struct file *filep,
+                                    unsigned long arg);
+static int lsm6dso32_get_info(FAR struct sensor_lowerhalf_s *lower,
+                              FAR struct file *filep,
+                              FAR struct sensor_device_info_s *info);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/* ODR frequencies to measurement intervals in microseconds */
+
+static const uint32_t ODR_INTERVAL[] =
+{
+  [ODR_OFF] = 0,      [ODR_12_5HZ] = 80000, [ODR_26HZ] = 38462,
```

Comment:
You shouldn't use designated initializers in common code. NuttX common code should be C89, and this feature is from C99. 

I know that some sensors use this approach but this is against NuttX standard and should be fixed some day: https://nuttx.apache.org/docs/latest/contributing/coding_style.html#initializers